### ISSUE TITLE
re-animation on re-render

### ISSUE THREAD

**Issue by mlp73** [glennreyes/react-countup#32]
Hello. It appears like a react re-render causes the countup to re-run. It is particularly visible when using two countup on the same render with one depending on an asynchronous call. The latest data arriving will cause both to re-render.

Any idea how to prevent that?

**Comment by glennreyes** [comment]
@mlp73 This component will start the animation every time the render method is called. One way to manage it would be to take advantage of the `shouldComponentUpdate()` [lifecycle method](https://facebook.github.io/react/docs/component-specs.html#updating-shouldcomponentupdate) of your higher order component.

e.g.:

```js
import React, { Component } from 'react';
import CountUp from 'react-countup';


class MyCountUp extends Component {
  shouldComponentUpdate = (nextProps, nextState) => {
    // return true if count up should animate
  }

  render() {
    <CountUp {....this.props} />
  }
}
```

**Comment by mlp73** [comment]
Countup is only a child, and therefore not the only element in the render method. Do you have an example of shouldcomponentUpdate only preventing a re-render on Countup without preventing other children?

**Comment by glennreyes** [comment]
@mlp73 Sure! Split up into seperate components.

e.g.:
```
const WrapperComponent () => (
  <div>
    <AnotherComponent />
    <MyCounter />
  </div>
);
```
Does that help?

Read also: https://facebook.github.io/react/docs/thinking-in-react.html

I'll close this issue as it's not count-up specific.

**Comment by mlp73** [comment]
haha thx yeah, but no this was not what I meant. ;)

Fair enough if you don't consider this is related to CountUp itself. But if you look at other animated components like chart js, you'll see that redraw is an option. I believe we should'nt have to systematically extend with a shouldComponentUpdate as I now had to, I mean this should be an option here too.

**Comment by glennreyes** [comment]
Yeah got it now, we could add this prop here too.
I would appreciate a PR for this!

**Comment by glennreyes** [comment]
This is now implemented, see #38
